sustainml_cpp/include/sustainml_cpp/orchestrator/OrchestratorNode.hpp

```diff
@@ -174,16 +173,26 @@ class OrchestratorNode
     /**
      * @brief Used to retrieve the associated OrchestratorNodeHandle.
      */
-    inline std::weak_ptr<OrchestratorNodeHandle> get_handler()
+    inline OrchestratorNodeHandle& get_handler()
```

Comment:
I would turn this into

```suggestion
    inline OrchestratorNodeHandle* get_handler() const
```

With the subsequent changes in tests